src/main/java/carpet/script/value/EntityValue.java

```diff
@@ -1319,8 +1344,24 @@ else if (v instanceof ListValue)
             if(e instanceof PlayerEntity) ((PlayerEntity) e).getHungerManager().addExhaustion((int) NumericValue.asNumber(v).getLong());
         });
 
+        put("absorption", (e, v) -> {
+            if (e instanceof PlayerEntity) ((PlayerEntity) e).setAbsorptionAmount((float) NumericValue.asNumber(v).getLong());
+        });
+
+        put("add_xp", (e, v) -> {
+            if (e instanceof PlayerEntity) ((PlayerEntity) e).addExperience(NumericValue.asNumber(v).getInt());
+        });
+
+        put("xp_level", (e, v) -> {
+            if (e instanceof PlayerEntity) ((PlayerEntity) e).addExperienceLevels(NumericValue.asNumber(v).getInt()-((PlayerEntity) e).experienceLevel);
+        });
+
+        put("xp_score", (e, v) -> {
```

Comment:
What about modify(.., 'xp_progress') and that function to computes points per level that I mentioned earlier?

Reply:
k

Reply:
butidk what exactly should It do? set the player's xp based on the fill level of xp bar?

Reply:
@gnembon can u explain, cos all I need to do is this before we can finally merge this age-old pr.